ci/integTestGen/src/integTestGen.jl

```diff
@@ -0,0 +1,265 @@
+module integTestGen
+
+import Pkg
+import PkgDependency
+import YAML
+
+"""
+    create_working_env(project_path::AbstractString)
+
+Create an temporary folder, setup a new Project.toml and activate it. Checking the dependencies of
+an project only works, if it is a dependency of the integTestGen.jl. Because the package to analyze
+is only a temporary dependency, it should not change permanent the Project.toml of the
+integTestGen.jl. Therefore, the script generates a temporary Julia environment and adds the package
+to analyze as dependency.
+
+# Args
+    `project_path::AbstractString`: Absolute path to the project folder of the package to analyse
+"""
+function create_working_env(project_path::AbstractString)
+    tmp_path = mktempdir()
+    Pkg.activate(tmp_path)
+    # same dependency like in the Project.toml of integTestGen.jl
+    Pkg.add("Pkg")
+    Pkg.add("PkgDependency")
+    Pkg.add("YAML")
+    # add main project as dependency
+    Pkg.develop(path=project_path)
+end
+
+"""
+Contains all information about a package.
+
+# Fields
+- `url`: Git url of the original project.
+- `modified_url`: Stores the Git url set by the environment variable.
+- `env_var`: Name of the environment variable to set the modified_url.
+"""
+mutable struct PackageInfo
+    url::String
+    modified_url::String
+    env_var::String
+    PackageInfo(url, env_var) = new(url, "", env_var)
+end
+
+package_infos = Dict(
```

Comment:
Mutable globals are unsafe! I think in this case there are fine, but maybe move them to the *execution block* beyond line 237.